iot_verifier/src/tx_scaler.rs

```diff
@@ -65,30 +74,31 @@ impl Server {
             }
 
             tokio::select! {
-                _ = trigger_timer.tick() => self.refresh_scaling_map().await?,
+                _ = trigger_timer.tick() => self.refresh_scaling_map(gateway_cache).await?,
                 _ = shutdown.clone() => return Ok(()),
             }
         }
     }
 
-    pub async fn refresh_scaling_map(&mut self) -> Result<(), TxScalerError> {
+    pub async fn refresh_scaling_map(
+        &mut self,
+        gateway_cache: &GatewayCache,
+    ) -> Result<(), TxScalerError> {
         let refresh_start = Utc::now();
         tracing::info!("density_scaler: generating hex scaling map, starting at {refresh_start:?}");
         let mut global_map = GlobalHexMap::new();
         let active_gateways = self
             .gateways_recent_activity(refresh_start)
             .await
             .map_err(sqlx::Error::from)?;
-        let mut gw_stream = self.follower.active_gateways().await?;
-        while let Some(GatewayInfo {
-            location, address, ..
-        }) = gw_stream.next().await
-        {
-            if let Some(h3index) = location {
-                if active_gateways.contains_key(&address) {
+        let mut gw_stream = self.iot_config_client.active_gateways().await?;
+        while let Some(gateway_info) = gw_stream.next().await {
+            if let Some(h3index) = gateway_info.location {
+                if active_gateways.contains_key(&gateway_info.address.as_ref().to_vec()) {
                     global_map.increment_unclipped(h3index)
                 }
             }
+            _ = gateway_cache.insert(gateway_info).await;
```

Comment:
i applaud the efficiency of using this operation to warm the cache even as i hate that it's happening in a module that's not _obviously_ supposed to be doing that 😆